client/src/components/Atoms/FileUpload.tsx

```diff
@@ -0,0 +1,156 @@
+import React from 'react'
+import { FileInput, Button, Colors, ProgressBar } from '@blueprintjs/core'
+import { useForm } from 'react-hook-form'
+import styled from 'styled-components'
+import StatusTag from './StatusTag'
+import { IFileUpload } from '../useFileUpload'
+import AsyncButton from './AsyncButton'
+
+const ErrorP = styled.p`
+  margin-top: 10px;
+  color: ${Colors.RED3};
+`
+
+export interface IFileUploadProps extends IFileUpload {
+  acceptFileType: 'csv' | 'zip'
+  allowMultipleFiles?: boolean
+  disabled?: boolean
+}
+
+const FileUpload = ({
+  uploadedFile,
+  uploadFiles,
+  uploadProgress,
+  deleteFile,
+  downloadFileUrl,
+  acceptFileType,
+  allowMultipleFiles = false,
+  disabled = false,
+}: IFileUploadProps) => {
+  const { register, handleSubmit, formState, watch } = useForm<{
+    files: FileList
+  }>({ mode: 'onTouched' })
+
+  if (!uploadedFile.isSuccess) return null
```

Comment:
Got it, that makes sense!

Of course out of scope for this PR but I do like having explicit loading state indicators to avoid a jitter as items disappear and reappear